### PR TITLE
[XdsClient] Add missing authority to XdsClient metrics scope

### PR DESCRIPTION
**Pull request by yashykt**
Fix for b/365993761.

Noticed that XdsClient metrics were not being reported due to authority not being properly set. 

This solution is not perfect since channels created later can possibly use a different authority, so preferring to use the default authority from the first channel.

### CODE REVIEW

File: src/core/xds/grpc/xds_client_grpc.cc [hunk 1]
```diff
@@ -233,6 +250,11 @@ absl::StatusOr<RefCountedPtr<GrpcXdsClient>> GrpcXdsClient::GetOrCreate(
   // instance for the channel or server instead of using the global instance.
   absl::optional<absl::string_view> bootstrap_config = args.GetString(
       GRPC_ARG_TEST_ONLY_DO_NOT_USE_IN_PROD_XDS_BOOTSTRAP_CONFIG);
+  std::string channel_default_authority =
```
Comment (markdroth) [inline src/core/xds/grpc/xds_client_grpc.cc:253]: Doesn't look like this is used anywhere, so I don't think it's needed.

File: src/core/xds/grpc/xds_client_grpc.cc [hunk 2]
```diff
@@ -225,6 +225,23 @@ absl::StatusOr<std::string> GetBootstrapContents(const char* fallback_config) {
...
 }
 
+GlobalStatsPluginRegistry::StatsPluginGroup
+GetStatsPluginGroupForKeyAndChannelArgs(absl::string_view key,
+                                        const ChannelArgs& channel_args) {
+  if (key == GrpcXdsClient::kServerKey) {
+    return GlobalStatsPluginRegistry::GetStatsPluginsForServer(channel_args);
+  }
+  grpc_event_engine::experimental::ChannelArgsEndpointConfig endpoint_config(
+      ChannelArgs{});
```
Comment (markdroth) [inline src/core/xds/grpc/xds_client_grpc.cc:235]: This needs to be changed to `channel_args` if we actually want those args to be used here.

File: test/cpp/end2end/xds/xds_core_end2end_test.cc [hunk 3]
```diff
@@ -1115,9 +1115,16 @@ TEST_P(XdsFederationTest, FederationServer) {
...
-                        .UseDisabledByDefaultMetrics(true)
-                        .BuildAndRegister();
+    stats_plugin_ =
+        grpc_core::FakeStatsPluginBuilder()
+            .UseDisabledByDefaultMetrics(true)
+            .SetChannelFilter(
+                [](const grpc_core::experimental::StatsPluginChannelScope&
+                       scope) {
+                  return scope.target() == absl::StrCat("xds:", kServerName) &&
+                         scope.default_authority() == kServerName;
```
Comment (markdroth) [inline test/cpp/end2end/xds/xds_core_end2end_test.cc:1125]: Should we also check the channel args here?

Comment (yashykt) [reply test/cpp/end2end/xds/xds_core_end2end_test.cc]: Done

File: src/core/xds/grpc/xds_client_grpc.cc (hunk 2 above)
Comment (yashykt) [reply src/core/xds/grpc/xds_client_grpc.cc]: Done

File: src/core/xds/grpc/xds_client_grpc.cc (hunk 1 above)
Comment (yashykt) [reply src/core/xds/grpc/xds_client_grpc.cc]: Done

File: src/core/xds/grpc/xds_client_grpc.cc [hunk 4]
```diff
@@ -232,7 +232,7 @@ GetStatsPluginGroupForKeyAndChannelArgs(absl::string_view key,
     return GlobalStatsPluginRegistry::GetStatsPluginsForServer(channel_args);
   }
   grpc_event_engine::experimental::ChannelArgsEndpointConfig endpoint_config(
-      ChannelArgs{});
+      ChannelArgs{channel_args});
```
Comment (markdroth) [inline src/core/xds/grpc/xds_client_grpc.cc:235]: No need for `ChannelArgs{}` around this -- we can just pass `channel_args` directly to the endpoint_config ctor.

**Review by markdroth — APPROVED** [review]
Just one more minor comment to fix; feel free to merge after addressing.

File: src/core/xds/grpc/xds_client_grpc.cc (hunk 4 above)
Comment (yashykt) [reply src/core/xds/grpc/xds_client_grpc.cc]: ofcourse :D

**Review by markdroth — APPROVED** [review]

**Pull request closed without merging**